glfwPlatformSupported returns false for X11 when DISPLAY is not set
diff -ruN a/src/platform.c b/src/platform.c
--- a/src/platform.c
+++ b/src/platform.c
@@ -113,6 +113,9 @@
     if (platformID == GLFW_PLATFORM_NULL)
         return GLFW_TRUE;
 
+    if (platformID == GLFW_PLATFORM_X11 && !getenv("DISPLAY"))
+        return GLFW_FALSE;
+
     for (i = 0;  i < count;  i++)
     {
         if (platformID == supportedPlatforms[i].ID)
